modules/line_descriptor/src/binary_descriptor.cpp

```diff
@@ -2353,12 +2353,59 @@ int BinaryDescriptor::EDLineDetector::EDline( cv::Mat &image, LineChains &lines
           double a5 = lineEqu[2] * lineEqu[1];
           unsigned int Px = pLineXCors[pLineSID[numOfLines]];         //first pixel
           unsigned int Py = pLineYCors[pLineSID[numOfLines]];
-          lineEndP[0] = (float) ( a1 * Px - a3 * Py - a4 );         //x
-          lineEndP[1] = (float) ( a2 * Py - a3 * Px - a5 );         //y
+		  float x1 = (float) ( a1 * Px - a3 * Py - a4 );         //x
+          float y1 = (float) ( a2 * Py - a3 * Px - a5 );         //y
           Px = pLineXCors[offsetInLineArray - 1];         //last pixel
           Py = pLineYCors[offsetInLineArray - 1];
-          lineEndP[2] = (float) ( a1 * Px - a3 * Py - a4 );         //x
-          lineEndP[3] = (float) ( a2 * Py - a3 * Px - a5 );         //y
+          float x2 = (float) ( a1 * Px - a3 * Py - a4 );         //x
+          float y2 = (float) ( a2 * Py - a3 * Px - a5 );         //y
+		  // clip line to image size. 
```

Comment:
please, make a function and call it, instead of repeating the clipping algorithm twice